kernel/src/hil/storage_interface.rs

```diff
@@ -0,0 +1,74 @@
+//! Interfaces for storage devices.
+
+use crate::returncode::ReturnCode;
+
+pub type StorageLen = usize;
+pub type OperationResult = Result<(), (ReturnCode, Option<&'static mut [u8]>)>;
+
+/// Cookies represent seekable positions within a storage interface. `SeekBeginning` allows a
+/// client to seek to the very beginning of the interface. `Cookie` allows the client to seek to a
+/// particular position within the interface. How a `Cookie` internally represents a position is up
+/// to the implementer.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum StorageCookie {
+    SeekBeginning,
+    Cookie(usize),
```

Comment:
What is the `usize`? Maybe the `Cookie { field: type }` version would be better.

Reply:
Just a value to represent some entry in the log, how it does so is left to the implementer. Would it make more sense to give this value a name?